src/data_factory_testing_framework/functions/_expression_transformer.py

```diff
@@ -98,131 +83,115 @@ def EXPRESSION_ARRAY_INDEX(self, token: Token) -> int:  # noqa: N802
+            parameter_name,
+        )
 
-    def expression_activity_reference(self, values: list[Token, str, int, float, bool]) -> [str, int, float, bool]:
-        if not (isinstance(values[0], Token) and values[0].type == "EXPRESSION_ACTIVITY_NAME"):
+    def expression_activity_reference(
+        self, values: list[Tree, Token, str, int, float, bool]
+    ) -> [str, int, float, bool]:
+        if len(values) != 2:
+            raise ExpressionEvaluationError("Activity reference should have two values")
+
+        expression_activity_name = values[0]
+        if not isinstance(expression_activity_name, Token):
+            raise ExpressionEvaluationError("Activity reference requires Token")
+
+        if not expression_activity_name.type == "EXPRESSION_ACTIVITY_NAME":
             raise ExpressionEvaluationError('Activity reference requires Token "EXPRESSION_ACTIVITY_NAME"')
 
-        activity_name = values[0].value
+        activity_name = expression_activity_name.value
         activity_name = activity_name[1:-1]  # remove quotes
 
-        if not all(
-            isinstance(value, Token) and value.type in ["EXPRESSION_PARAMETER_NAME", "EXPRESSION_ARRAY_INDEX"]
-            for value in values[1:]
-        ):
-            raise ExpressionEvaluationError(
-                'Activity property and fields should be of type "EXPRESSION_PARAMETER_NAME" or "EXPRESSION_ARRAY_INDEX"'
-            )
+        activity = self.state.get_activity_result_by_name(activity_name)
+
+        return self._evaluate_expression_object_accessors(activity, [values[1]])
 
-        activity = self.state.try_get_scoped_activity_result_by_name(activity_name)
-        if activity is None:
-            raise ActivityNotFoundError(activity_name)
-
-        property_tokens = values[1:]
-        current_output = activity
-
-        while len(property_tokens) > 0:
-            current_token = property_tokens.pop(0)
-            if current_token.type == "EXPRESSION_PARAMETER_NAME":
-                if current_token.value not in current_output:
-                    raise ActivityOutputFieldNotFoundError(activity_name, current_token.value)
-
-                current_output = current_output[current_token.value]
-                continue
-            if current_token.type == "EXPRESSION_ARRAY_INDEX":
-                if not isinstance(current_output, list):
-                    raise ExpressionEvaluationError("Array index can only be used on lists")
-                current_output = current_output[current_token.value]
-                continue
-        return current_output
-
-    def expression_item_reference(self, value: list[Token, str, int, float, bool]) -> [str, int, float, bool]:
+    def expression_item_reference(self, values: list[Tree, Token, str, int, float, bool]) -> [str, int, float, bool]:
         item = self.state.iteration_item
         if item is None:
             raise StateIterationItemNotSetError()
+
+        if len(values) != 0:
```

Comment:
Consider moving the guard clause to the top of the method body.

Reply:
done!